### PR TITLE
(Calendar) Add Thai DateFormat Supports

### PR DESCRIPTION
**Pull request by cdmochi**
I wanted supports for buddhist calendar in Thai. This pull request contains localization and buddhist year supported.

### CODE REVIEW

**Comment by maxkeppeler** [comment]
Hey, I would love to see the support for Buddhist calendar and see your PR merged!

**Comment by cdmochi** [comment]
I would love to see any suggestions regarding to my PR 😀.

**Review by maxkeppeler — CHANGES_REQUESTED** [review]

File: calendar/src/main/java/com/maxkeppeler/sheets/calendar/CalendarState.kt [hunk 1]
```diff
@@ -185,7 +211,12 @@ internal class CalendarState(
     }
 
     fun onYearClick(year: Int) {
-        var newDate = cameraDate.withYear(year)
+        var newDate = if (config.locale.language == Locale("th").language) {
+            cameraDate.withYear(year - 543)
+        } else {
+            cameraDate.withYear(year)
+        }
+
```
[the remark this answers is not in the document]
Comment (cdmochi) [reply calendar/src/main/java/com/maxkeppeler/sheets/calendar/CalendarState.kt:219]: Firstly, line 214 the year in calendarView will be shown as Thai Buddhism year format if and only if Thai locale is set. 
Also, To calculate BE year manually, just add 543 to year to the BC year. You may look for more references here from the link I provided below as well. [https://wesak.org.my/calculating-b-e/#:~:text=Thus%20543%20BCE%2C%20the%20year,B%20%3D%20544%20%2B%202021%20%3D%20B.E.](url)

[the remark this answers is not in the document]
Comment (cdmochi) [reply calendar/src/main/java/com/maxkeppeler/sheets/calendar/CalendarState.kt:219]: > What does that do? Why do you have to check for locale th when clicking a year and why do you reduce the selected year amount with 543 years?

Firstly, line 214 the year in calendarView will be shown as Thai Buddhism year format if and only if Thai locale is set.

[the remark this answers is not in the document]
Comment (cdmochi) [reply calendar/src/main/java/com/maxkeppeler/sheets/calendar/CalendarState.kt:219]: > Should we consider creating a separate utility function within the utils file? I would appreciate your thoughts on this.

To calculate BE year manually, just add 543 to year to the BC year. You may look for more references here from the link I provided below as well. [https://wesak.org.my/calculating-b-e/#:~:text=Thus%20543%20BCE%2C%20the%20year,B%20%3D%20544%20%2B%202021%20%3D%20B.E.](https://github.com/maxkeppeler/sheets-compose-dialogs/pull/url)

[the remark this answers is not in the document]
Comment (cdmochi) [reply calendar/src/main/java/com/maxkeppeler/sheets/calendar/CalendarState.kt:219]: > Should we consider creating a separate utility function within the utils file? I would appreciate your thoughts on this.

I think that would make it better in this case. I'll fix it.

File: calendar/src/main/java/com/maxkeppeler/sheets/calendar/views/YearSelectionView.kt [hunk 2]
```diff
@@ -27,10 +29,16 @@ import java.time.LocalDate
...
+    config: CalendarConfig
 ) {
     items(yearsRange.endInclusive.minus(yearsRange.start).plus(1)) {
-        val year = yearsRange.start + it
+        val year = if (config.locale.language == Locale("th").language) {
+            (yearsRange.start + it) + 543
+        } else {
+            yearsRange.start + it
+        }
+
```
[the remark this answers is not in the document]
Comment (cdmochi) [reply calendar/src/main/java/com/maxkeppeler/sheets/calendar/views/YearSelectionView.kt:41]: I think so, i'll fix it

**Comment by maxkeppeler** [comment]
Closed due to inactivity

**Pull request closed without merging**